### PR TITLE
Power laser port 2 electric bogaloo

### PR DESCRIPTION
**Pull request by Migratingcocofruit**
## What Does This PR Do
Ports the power transmission laser again, this time with 50% more description and 100% less broken behavior(let me dream okay).

The Power Transmission Laser sells the excess power produced by the station for credits.
Credits earned per cycle approach 0 as the output power approaches 0, and approaches 10 as the power output approaches infinity.
75% of the money generated goes to engineering, and the remaining 25% go to cargo. 
The PTL will send money when it reaches 200 credits of stored funds so it doesn't spam the accounts with tiny transactions

The PTL will send updates as the station sends more and more energy to central, the threshold for a new announcement rises exponentially capped at 200GJ above the last one.

The PTL produces a beam that is purely visual now

The bore of the PTL will produce flashes from 5 MW of output power and up. The flash strength is output power in megawatts divided by 5 and capped by 3. Looking directly at the bore(opposite the direction of the PTL) gives the full flash effect, looking away from it gives no flash, and looking at it partly multiplies the flash strength by the cosine of the difference between the direction you're looking at and a line from you to the bore. The range scales with power in the same way and caps at 8 tiles away, and you can't get flashed while standing behind the bore(indicated by a black and yellow line in the PTL room)

Radiation starts at 30MW with the pulse strength being 50 times output power in megawatts divided by 30. This isn't capped, but should produce only a moderate amount of radiation even when outputting a gigawatt.

The PTL can be used to target megafauna in the mining base and it's immediate vicinity, dealing 10 * output in megawatts fire damage. 
Megafauna damaged by the laser lose their loot, and megafauna that are killed by the laser also get dusted.
Once the targeted megafauna is dead the laser will automatically revert to selling power to central.

At the end of the round the total amount of energy exported by the PTL(in GigaJoules) is presented

PTL locations:

Cyberiad
image

Kerberos
image

Cerebron
image

Farragus
image

Diagoras
image

## Why It's Good For The Game
Allows engineers to do something with all the power they make.

## Images of changes
380451347-0a939632-d3f6-4bbb-8d63-fb12a1022d36
image

## Testing
So far:
- Observed the correct amount of payout being given at the correct threshold.
- Ordered the PTL crate and used it to build one
- Rotated the PTL and made sure it rotates when it should, and not when it shouldn't
- Spawned lavaland mobs in the mining base and blasted them. They did not drop loot when they died.
- Got flashed and irradiated by the PTL

<hr>

## ToDo
- [x] Make the blocker ref into a stored UID value.
- [x] Make fine adjustment better for low power use
- [x] Have max capacity not be infinite
- [x] Make the laser deconstructable and give it components and a cargo crate.
- [x] Add missing stuff to the rooms
- [x] Make the rooms not ugly
- [x] Get the PTL to show up properly on the map editor
- [x] Implement bombardment of megafauna close to the mining base
- [x] Delete megafauna loot if they are killed by the laser

### Declaration
image

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl:
add: Ports the power transmission laser, a machine that sells the excess power produced by the station for credits
/:cl:

### CODE REVIEW

**Comment by Henri215** [comment]
Implement the fixes from https://github.com/ParadiseSS13/Paradise/pull/27167 and https://github.com/ParadiseSS13/Paradise/pull/27169

**Comment by Migratingcocofruit** [comment]
> Implement the fixes from #27167 and #27169

Already did

**Comment by Henri215** [comment]
> > Implement the fixes from #27167 and #27169
> 
> Already did

And you did commit it? Because i am seeing the code from before the fixes.

File: code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+	return get_step(center, dir)
+
+
+
+/obj/machinery/power/transmission_laser/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Laser currently has [unsent_earnings] unsent credits.\s<span/>"
+	. += "<span class='notice'>Laser has generated [total_earnings] credits.\s<span/>"
+	. += "<span class='notice'>Laser has sold [total_energy] Joules\s<span/>"
+
```
Comment (Henri215) [inline code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:178]:
```suggestion
	. += "<span class='notice'>Laser currently has [unsent_earnings] unsent credits.<span/>"
	. += "<span class='notice'>Laser has generated [total_earnings] credits.<span/>"
	. += "<span class='notice'>Laser has sold [total_energy] Joules.<span/>"
```
Theres this one from the fix too.

**Review by Henri215 — COMMENTED** [review]
The others will be sending more stuff to be fixed soon.

File: code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm (hunk 1 above)
Comment (Migratingcocofruit) [reply code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:178]: done

**Comment by S34NW** [comment]
Here was some feedback I was given by dev team members, apologies if anything duplicates what you've done:

- You're using both pixel_y/pixel_x and the multitile component. You can make it so you only need the multitile component
- The `var/static/announcement_threshold` can probably be  changed to a normal variable
- `var/atom/blocker` should really be a UID instead of a ref.
- get_back_turf() isn't necessary with a good multitile component
- This should be rebuildable/deconstructable in case of repairs
- Ghosts cant view the TGUI
- beams through cerestation disposals leading to items either being heavily damaged or you know gibbing disposaled people
- The fact it is a beam that makes space travel around it impossible (without changing zlvls) rather than just beaming the power through the roof.
- The map roomed are silly.
- Machine is indestructible.
- Mapped in roundstart (preference)
- Doesn't look proper in the map editor (also linked to the issue that it doesnt properly use multitile)
- Its already mapped in, I believe most people thought it would be something you had to build
- Money is kinda underwhelming for how much you have to put into it
- Gib + explosion laser is too powerful
-  the knob hardly moves <10mw
delwedd
- you can set the 'output' power to whatever you want, and insta gib/destroy any items that're in the path, even if it lacks the power to do so
- You can do funny things like this (it doesn't generate any more cash, and will automatically shut down but the beam will vaporise you)
delwedd
- the new rooms are giant bare boxes with no decorations or contents. if we're adding an entirely new engineering area, it needs to be thematic, it needs to look like it belongs in engineering. what items should be in it? any lockers? storage? warning signs on the walls, warning decals around the laser?
- it almost certainly doesn't need reinforced walls.
- on cere it needs to be put somewhere it won't affect the cere waste loop, even if that means altering the waste loop path or putting the laser further out and adding disposals pipes where the new area intersects the waste loop
- the room needs to be more structurally integrated into the rest of the station. the overall shape of the perimeter should be followed more closely. i know box gravgen is another big boxy room but that sucks too and we don't want to keep making it worse
- fire alarms are missing on box and cere
- there's no cameras

**Comment by 1080pCat** [comment]
> * The fact it is a beam that makes space travel around it impossible (without changing zlvls) rather than just beaming the power through the roof.

i quite like this, it makes space more risky, and requires communication. however the PTL mirrors should be added/ported so the beam can be snaked around, so it doesn't need to be in a odd spot.

**Comment by 1080pCat** [comment]
Another thing. I personally think the how the BSH handled the power level/input before you reworked it would be the best thing for handling the power
image

Having a slider bar go from 0-1000 (kw, mw, gw) with the green bar showing the grid saturation/reserve energy would be nice.
(more of a wiki thing than anything) but the reserve power makes /no sense/ to me, i could never get the bar to move, the GJ would go up very slowly, but the bar would never move. I dunno if this was a bug, or just misunderstood mechanics.

**Comment by Migratingcocofruit** [comment]
> Here was some feedback I was given by dev team members, apologies if anything duplicates what you've done:
> 
>     * You're using both pixel_y/pixel_x and the multitile component. You can make it so you only need the multitile component

pixel_x and pixel_y move the sprite while the multitile component adds invisible dense structures that only impede movement. The BSH and DNA vault also do this. 
The reason it doesn't look right in the map editor is because the machine itself is actually only one tile, and that tile needs a wire node under it. 
The offsets are used so that the machine could be rotated around that tile, preventing it from getting disconnected from the powernet if it is rotated for any reason.

>     * The `var/static/announcement_threshold` can probably be  changed to a normal variable
I was asked to make it a static in the previous PR. And considering the announcement are about the power sent to centcom, rather than the power output by the PTL it makes sense that they would be shared between them.

>     * `var/atom/blocker` should really be a UID instead of a ref.
Yeah, I think I can make it work.
> 
>     * get_back_turf() isn't necessary with a good multitile component
> 
Get back turf isn't called at all. I forgot to delete it.

>     * This should be rebuildable/deconstructable in case of repairs
> 
I'll make it so. And make a cargo crate for it as well.

>     * Ghosts cant view the TGUI
> 
Fixed that already
>     * beams through cerestation disposals leading to items either being heavily damaged or you know gibbing disposaled people
> 
Oh yeah. I'll take care of that.
>     * The fact it is a beam that makes space travel around it impossible (without changing zlvls) rather than just beaming the power through the roof.
> 
I'll get the mirrors as well, or maybe just have it done with reflectors.

>     * The map roomed are silly.
> 
:(
>     * Machine is indestructible.
>
duplicate.  
>     * Mapped in roundstart (preference)
> 
I'd rather have it up at roundstart. This is a good default project for engineering and I'd rather it requires as little effort to setup as possible, with most of the effort being put into power production. Minimum effort credits will likely need adjustment, but I think playtesting would be the best way to figure it out.
>     * Doesn't look proper in the map editor (also linked to the issue that it doesnt properly use multitile)
> 
It does use multitile properly. Again it's due to having it rotate around the wire node under it.

>     * Its already mapped in, I believe most people thought it would be something you had to build
> 
Again, duplicate.
>     * Money is kinda underwhelming for how much you have to put into it
> 
You can get around 7 credits per second with a powerfull SM setup. This doesn't sound like much, but it comes up to 25,200 credits per hour.
>     * Gib + explosion laser is too powerful
> 
I don't really know how powerfull the laser effects should be in the end and at what thresholds. Open to suggestions on that
>     * the knob hardly moves <10mw

Are logarithmic knobs a thing? I'll try to figure something out.

>     * you can set the 'output' power to whatever you want, and insta gib/destroy any items that're in the path, even if it lacks the power to do so
> 
>     * You can do funny things like this (it doesn't generate any more cash, and will automatically shut down but the beam will vaporise you)

Fixed that 

>     * the new rooms are giant bare boxes with no decorations or contents. if we're adding an entirely new engineering area, it needs to be thematic, it needs to look like it belongs in engineering. what items should be in it? any lockers? storage? warning signs on the walls, warning decals around the laser?
> 
>     * it almost certainly doesn't need reinforced walls.
> 
>     * on cere it needs to be put somewhere it won't affect the cere waste loop, even if that means altering the waste loop path or putting the laser further out and adding disposals pipes where the new area intersects the waste loop
> 
>     * the room needs to be more structurally integrated into the rest of the station. the overall shape of the perimeter should be followed more closely. i know box gravgen is another big boxy room but that sucks too and we don't want to keep making it worse
> 
>     * fire alarms are missing on box and cere
> 
>     * there's no cameras

I'll fix the obvious deficiencies now, but I may need some guidance in regards to decoration and overall aesthetic design.

I'll make a to-do list in the PR comment for all the stuff.

**Comment by Migratingcocofruit** [comment]
> Another thing. I personally think the how the BSH handled the power level/input before you reworked it would be the best thing for handling the power image
> 
> Having a slider bar go from 0-1000 (kw, mw, gw) with the green bar showing the grid saturation/reserve energy would be nice. (more of a wiki thing than anything) but the reserve power makes /no sense/ to me, i could never get the bar to move, the GJ would go up very slowly, but the bar would never move. I dunno if this was a bug, or just misunderstood mechanics.

That's because the maximum value is infinity. I need to figure out where to put the limit on that as well, I suppose.

**Comment by Henri215** [comment]
"add: Ports the power transmission laser, a machinery that sells the excess power produced by the station for credits"

This changelog would give the players more information.

**Comment by Migratingcocofruit** [comment]
Reflection is deemed problematic, so I won't implement it.

File: code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+	announcer = new(config_type = /datum/announcement_configuration/ptl)
+	range = get_dist(get_front_turf(), get_edge_target_turf(get_front_turf(), dir))
+	if(!powernet)
+		connect_to_network()
+	handle_offset()
+	update_icon()
+
+/obj/machinery/power/transmission_laser/screwdriver_act(mob/living/user, obj/item/I)
+	if(firing)
+		to_chat(user,"<span class='info'>Turn the laser off first<span/>")
```
Comment (Henri215) [inline code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:114]:
```suggestion
		to_chat(user,"<span class='info'>Turn the laser off first.<span/>")
```

File: code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+
+		last_step = get_step(last_step, dir)
+
+/obj/machinery/power/transmission_laser/proc/destroy_lasers()
+	for(var/obj/effect/transmission_beam/listed_beam as anything in laser_effects)
+		laser_effects -= listed_beam
+		qdel(listed_beam)
+
+// Beam
+/obj/effect/transmission_beam
```
Comment (Henri215) [inline code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:443]: It cant be a /obj/effect/ebeam for what you want to do with it?

Comment (Migratingcocofruit) [reply code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:443]: No. Following a discussion with balance and design the beam is going to be a purely visual effect that goes under everything. Making it an ebeam will only make it more complicated.

**Review by warriorstar-orion — CHANGES_REQUESTED** [review]
Map changes looking good.

## Cere

Move these airlocks down so that the waste loop area is walled off completely. Also move/remove this floating lightbulb:

image

Extra camera, use plating under window spawners, light tube embedded in wall, use flooring under airlocks, floating light bulb in front of first set of airlocks, move grille away from in front of first set of airlocks:

image

Pixel-shift these signs onto the wall from the tile in front of them, since they can't be seen from the other side:

image

Move the gap in the waste belt to directly in front of the PTL beam path:

image

The perimeter of any rock walls that can be dug directly to space need to be of type `/turf/simulated/mineral/ancient/outer`:

image

## Delta

The walling here is a little awkward; you want to avoid double-walling (in pink), and this gap in space feels out of place (in blue):

image

I would pull both of these to the adjacent walls, and use the extra space to decorate a bit more, something along these lines:

2024_10_26__11_27_37__paradise dme  deltastation dmm  - StrongDMM

Duplicate cameras on this station too.

## Meta

Use plating under window spawners, and push these vents/scrubbers so that they're not directly under lights. Or move the lights.

image

## Emerald

Floating light tube in front of airlock and in front of rack, use plating under window spawners, and move vents/scrubbers away from directly under lights.

image

These walls and windows bumping up against each other need to be adjusted. Either pull the rwall towards the airlock and remove the line of windows, or vice-versa.

image

Either remove this camera (these rooms don't need two cameras), or remove the other one and move this off the wall that has the APC on it.

## General

These areas would still benefit from light decoration in the form of colored tiles and decals. This is an idea how you may want to decorate things, but get creative:

2024_10_26__10_53_39__paradise dme  boxstation dmm  - StrongDMM

Especially note the station-specific theming. On Box, engineering doesn't use colored tiles extensively, but they are used directly surrounding airlocks.

**Comment by Migratingcocofruit** [comment]
> Map changes looking good.
> ## Cere
> 
> Move these airlocks down so that the waste loop area is walled off completely. Also move/remove this floating lightbulb:
> 
> image
> 
> Extra camera, use plating under window spawners, light tube embedded in wall, use flooring under airlocks, floating light bulb in front of first set of airlocks, move grille away from in front of first set of airlocks:
> 
> image
> 
> Pixel-shift these signs onto the wall from the tile in front of them, since they can't be seen from the other side:
> 
> image
> 
> Move the gap in the waste belt to directly in front of the PTL beam path:
> 
> image
> 
> The perimeter of any rock walls that can be dug directly to space need to be of type `/turf/simulated/mineral/ancient/outer`:
> 
> image
> ## Delta
> 
> The walling here is a little awkward; you want to avoid double-walling (in pink), and this gap in space feels out of place (in blue):
> 
> image
> 
> I would pull both of these to the adjacent walls, and use the extra space to decorate a bit more, something along these lines:
> 
> 2024_10_26__11_27_37__paradise dme deltastation dmm - StrongDMM
> 
> Duplicate cameras on this station too.
> ## Meta
> 
> Use plating under window spawners, and push these vents/scrubbers so that they're not directly under lights. Or move the lights.
> 
> image
> ## Emerald
> 
> Floating light tube in front of airlock and in front of rack, use plating under window spawners, and move vents/scrubbers away from directly under lights.
> 
> image
> 
> These walls and windows bumping up against each other need to be adjusted. Either pull the rwall towards the airlock and remove the line of windows, or vice-versa.
> 
> image
> 
> Either remove this camera (these rooms don't need two cameras), or remove the other one and move this off the wall that has the APC on it.
> ## General
> 
> These areas would still benefit from light decoration in the form of colored tiles and decals. This is an idea how you may want to decorate things, but get creative:
> 
> 2024_10_26__10_53_39__paradise dme boxstation dmm - StrongDMM
> 
> Especially note the station-specific theming. On Box, engineering doesn't use colored tiles extensively, but they are used directly surrounding airlocks.

Faragus changes pushed

**Comment by Migratingcocofruit** [comment]
> Map changes looking good.

> ## Delta
> 
> The walling here is a little awkward; you want to avoid double-walling (in pink), and this gap in space feels out of place (in blue):
> 
> image
> 
> I would pull both of these to the adjacent walls, and use the extra space to decorate a bit more, something along these lines:
> 
> 2024_10_26__11_27_37__paradise dme deltastation dmm - StrongDMM
> 
> Duplicate cameras on this station too.

Delta Changes pushed

**Comment by Migratingcocofruit** [comment]
Diagoras changes pushed

**Comment by Migratingcocofruit** [comment]
Meta changes pushed

**Comment by Migratingcocofruit** [comment]
Cyberiad changes pushed

**Review by warriorstar-orion — CHANGES_REQUESTED** [review]
Map changes look great. I like the visual distinctiveness of each station's room in terms of decorations.

On Cere:

Scrubber pipe extends out here:

2024_10_28__01_04_32__paradise dme  cerestation dmm  - StrongDMM

Catwalk hidden under this rock wall:

2024_10_28__01_02_39__paradise dme  cerestation dmm  - StrongDMM

All of these areas need to be updated to align back up with the substation and halls:

2024_10_28__01_02_22__paradise dme  cerestation dmm  - StrongDMM

Stray unexplored mining area in the hall:

2024_10_28__01_02_10__paradise dme  cerestation dmm  - StrongDMM

**Comment by Migratingcocofruit** [comment]
> Map changes look great. I like the visual distinctiveness of each station's room in terms of decorations.
> 
> On Cere:
> 
> Scrubber pipe extends out here:
> 
> All of these areas need to be updated to align back up with the substation and halls:
> 
> Stray unexplored mining area in the hall:

Done

**Comment by Onyxarias** [comment]
Currently placing a soft objection on this. The PR looks great from a Design point of view, but I would rather it not have so many broadcasted messages. Every other station goal doesn't do this except for sure emergencies. Instead, I'd like to see the messages restricted down to, say, just megafauna kills. Great work, by the way!

**Comment by Migratingcocofruit** [comment]
> Currently placing a soft objection on this. The PR looks great from a Design point of view, but I would rather it not have so many broadcasted messages. Every other station goal doesn't do this except for sure emergencies. Instead, I'd like to see the messages restricted down to, say, just megafauna kills. Great work, by the way!

I get what you mean, but I still want engineering to be able to show off. This is part of the point. Would changing the nature of the announcement be enough?

**Comment by Onyxarias** [comment]
> > Currently placing a soft objection on this. The PR looks great from a Design point of view, but I would rather it not have so many broadcasted messages. Every other station goal doesn't do this except for sure emergencies. Instead, I'd like to see the messages restricted down to, say, just megafauna kills. Great work, by the way!
> 
> I get what you mean, but I still want engineering to be able to show off. This is part of the point. Would changing the nature of the announcement be enough?

While true, myself and others have found the constant messages rather annoying, and redundant. If you compare it to another engineering machine, the Supermatter, even that only outputs messages during critical emergencies or anomalies, and the latter is only on the engineering channel. As well, Cargo doesn't have computers loudhailing when they sell certain amounts of plasma.

If engineering REALLY wants to show off, the CE can use the station announcements to relay their accomplishments to the station; it's not something that needs to be done automatically and _constantly._

Again, PR is great, and even with my objection will probably become another fantastic feature for a department that sorely needs it. But I don't think spamming the station's radio is a thing it should do.

**Comment by Migratingcocofruit** [comment]
> > > Currently placing a soft objection on this. The PR looks great from a Design point of view, but I would rather it not have so many broadcasted messages. Every other station goal doesn't do this except for sure emergencies. Instead, I'd like to see the messages restricted down to, say, just megafauna kills. Great work, by the way!
> > 
> > 
> > I get what you mean, but I still want engineering to be able to show off. This is part of the point. Would changing the nature of the announcement be enough?
> 
> While true, myself and others have found the constant messages rather annoying, and redundant. If you compare it to another engineering machine, the Supermatter, even that only outputs messages during critical emergencies or anomalies, and the latter is only on the engineering channel. As well, Cargo doesn't have computers loudhailing when they sell certain amounts of plasma.
> 
> If engineering REALLY wants to show off, the CE can use the station announcements to relay their accomplishments to the station; it's not something that needs to be done automatically and _constantly._
> 
> Again, PR is great, and even with my objection will probably become another fantastic feature for a department that sorely needs it. But I don't think spamming the station's radio is a thing it should do.

If everything else was also doing it the radio would be unusable. Good point, that has me convinced. I'll be axing the announcements, and give some more info on the UI.

File: code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+		return
+	if(default_deconstruction_crowbar(user, I))
+		return TRUE
+
+/obj/machinery/power/transmission_laser/wrench_act(mob/living/user, obj/item/I)
+	if(!panel_open)
+		return
+	if(rotate())
+		return TRUE
+	to_chat(user,"<span class='info'>Target area blocked, please clear all objects and personnel.<span/>")
```
Comment (Burzah) [inline code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:137]:
```suggestion
	to_chat(user, "<span class='info'>Target area blocked, please clear all objects and personnel.</span>")
```

File: code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm [hunk 5]
```diff
@@ -0,0 +1,537 @@
...
+	var/turf/center = locate(x + 1 + round(pixel_x / 32), y + 1 + round(pixel_y / 32), z)
+	return get_step(center, dir)
+
+
+
+/obj/machinery/power/transmission_laser/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Laser currently has [unsent_earnings] unsent credits.<span/>"
+	. += "<span class='notice'>Laser has generated [total_earnings] credits.<span/>"
+	. += "<span class='notice'>Laser has sold [total_energy] Joules.<span/>"
```
Comment (Burzah) [inline code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:217]:
```suggestion
	. += "<span class='notice'>Laser currently has [unsent_earnings] unsent credits.</span>"
	. += "<span class='notice'>Laser has generated [total_earnings] credits.</span>"
	. += "<span class='notice'>Laser has sold [total_energy] Joules.</span>"
```

File: code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+	component_parts += new /obj/item/stock_parts/capacitor
+	range = get_dist(get_front_turf(), get_edge_target_turf(get_front_turf(), dir))
+	if(!powernet)
+		connect_to_network()
+	handle_offset()
+	update_icon()
+
+/obj/machinery/power/transmission_laser/screwdriver_act(mob/living/user, obj/item/I)
+	if(firing)
+		to_chat(user,"<span class='info'>Turn the laser off first.<span/>")
```
[the remark this answers is not in the document]
Comment (Migratingcocofruit) [reply code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm]: done

File: code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm [hunk 7]
```diff
@@ -0,0 +1,537 @@
...
+	var/input_number = 0
+	/// Gives our power output when multiplied with power_format_multi_output. The multiplier signifies the units of power, and this is how many of them we are outputting.
+	var/output_number = 0
+	/// Our set input pulling
+	var/input_pulling = 0
+	/// Targetable areas in lavaland
+	var/list/targetable_areas = list(/area/lavaland/surface/outdoors/outpost,
+									/area/lavaland/surface/outdoors/targetable,
+									/area/mine/outpost,
+									/area/shuttle/mining)
```
[the remark this answers is not in the document]
Comment (Migratingcocofruit) [reply code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm]: done

File: code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm [hunk 8]
```diff
@@ -0,0 +1,537 @@
...
+	dir = WEST
+
+/obj/item/circuitboard/machine/transmission_laser
+	board_name = "Power Transmission Laser"
+	icon_state = "command"
+	build_path = /obj/machinery/power/transmission_laser
+	origin_tech = "engineering=2;combat=3;"
+	req_components = list(
+							/obj/item/stock_parts/capacitor = 3,
+							/obj/item/stock_parts/micro_laser = 3)
```
[the remark this answers is not in the document]
Comment (Migratingcocofruit) [reply code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm]: done

File: code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm (hunk 4 above)
Comment (Migratingcocofruit) [reply code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm]: done

File: code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm (hunk 5 above)
Comment (Migratingcocofruit) [reply code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm]: done

**Review by Burzah — APPROVED** [review]

**Pull request merged**